config/config_test.go

```diff
@@ -0,0 +1,168 @@
+package config
```

Comment:
When I'm building some new project--that is, where I'm doing exploratory coding because I don't yet know exactly what the project does or what it'll look like--I usually do the initial testing in the exact way the user will use that project. For example, for a tool you run manually on the command line (e.g. boilerplate), I'll test it manually on the command line. For a tool you use manually via a web UI (e.g. Atomic), I'll test it manually via the web UI. For a tool you use via an API (e.g. terratest), I'll test via programmatic tests. Of these, only the very last one is what purists would consider true TDD. However, I strongly believe in starting from the user's perspective and working backwards, and in many cases, TDD is not an accurate representation of the user's perspective.

Once the project looks right from the user's perspective, then I go back and clean up the code and fill in any missing automated tests. After that, I typically use TDD for all bug fixes and minor changes. However, large new features that require exploratory coding will follow the same pattern as above.